pyuff/datasets/dataset_2420.py

```diff
@@ -17,13 +17,14 @@ def _write2420(fh, dset):
         fh.write('%10i\n' % (dset['part_UID']))
         fh.write('%-80s\n' % (dset['part_name']))
 
-        for node in range(len(dset['nodes'])):
-            fh.write('%10i%10i%10i\n' % (dset['nodes'][node], dset['cs_type'], dset['cs_color']))
-            fh.write('CS%i\n' % dset['nodes'][node])
-            fh.write('%25.16e%25.16e%25.16e\n' % tuple(dset['local_cs'][node * 4, :]))
-            fh.write('%25.16e%25.16e%25.16e\n' % tuple(dset['local_cs'][node * 4 + 1, :]))
-            fh.write('%25.16e%25.16e%25.16e\n' % tuple(dset['local_cs'][node * 4 + 2, :]))
-            fh.write('%25.16e%25.16e%25.16e\n' % tuple(dset['local_cs'][node * 4 + 3, :]))
+        num_CS = min( (len(dset['CS_sys_labels']),  len(dset['CS_types']),  len(dset['CS_colors']),  len(dset['CS_names']), len(dset['CS_matrices'])) )
```

Comment:
Should not this lengths be equal?

Reply:
In theory, yes. But if some problems should occur I thought it might be the best if the program imported only the datasets that are complete. But you're right, it could be that something is missing "in between", so all data after the one where one of the fields is missing is corrupted.
Should I implement a check that tests if those values are identical and otherwise throws an exception or asserts?

Reply:
Yes, I think an exception would be better here!